ui/app/AppLayouts/Onboarding2/OnboardingLayout.qml

```diff
@@ -89,6 +90,13 @@ Page {
         readonly property bool backAvailable:
             stack.currentItem ? (stack.currentItem.backAvailableHint ?? true)
                               : false
+
+        
+        onCurrentItemChanged: {
+            if (!!stack.currentItem && !!stack.currentItem.pageName) {
+                root.pageNavigated(stack.currentItem.pageName)
```

Comment:
I think @micieslak just removed the `pageName` :laughing:

Reply:
Ah, you re-added it... there should be a way to extract it from the class/instance name imo

Reply:
The only way I saw to do it was to use `stack.currentItem.toString()` and then run a regex on it to isolate the class name.
That sounded quite dirty and I read that the format could change from one OS to the other, so I preferred to stick to something that we know will work 100% of the time.

Let me know if there is a way to get it cleanly that I missed.

Reply:
Alright, I also wanted to use the automatic version, but I read it could be bad so I opted for the safe version, but since both of you are fine with it, I'm totally ok using it, less job to do :smile: 

I also didn't know we had a utility for it. I'll move it to the Utils file.